### PR TITLE
refactor: change herald config schema

### PR DESCRIPTION
**Pull request by destifo**
- change herald config to more linear structure(less nested values)

### CODE REVIEW

File: herald.yaml [hunk 1]
```diff
@@ -13,6 +14,14 @@ task_store_backend:
   credentials:
     accessKeyId: minio
     secretAccessKey: password
+
+service_accounts:
+  - name: "system:serviceaccount:dev-s3-herald:default"
+    buckets:
```
Comment (zifeo) [inline herald.yaml:20]: I like it! As a future improvement, the bucket list could use a regex with a {namespace} and {serviceaccount} tag so the structure could be "dynamic", e.g. `{serviceaccount}-.*` let the service account do anything with the `dev-s3-herald-` prefix

File: herald.yaml [hunk 2]
```diff
@@ -92,26 +77,49 @@ buckets:
...
+        accessKeyId: minio
+        secretAccessKey: password
+  - name: swift-swift-replica
+    backend: openstack_swift
+    config:
+      auth_url: https://api.pub1.infomaniak.cloud/identity/v3
+      credentials:
+        username: "test:tester"
+        password: "testing"
+        project_name: "PCP-RP63UPV"
```
Comment (zifeo) [inline herald.yaml:99]: are the project name fixed when using the dev container or do they change?

**Review by zifeo — DISMISSED** [review]

File: src/backends/mirror.ts [hunk 3]
```diff
@@ -68,8 +70,27 @@ export async function prepareMirrorRequests(
...
+  for (const replica of globalConfig.replicas) {
+    let primaryBucket = "";
+    if (bucketConfig.typ === "S3BucketConfig") {
+      primaryBucket = bucketConfig.config.bucket;
+    } else {
+      primaryBucket = bucketConfig.config.container;
+    }
+
+    let replicaBucket = "";
+    if (replica.typ === "ReplicaS3Config") {
```
Comment (zifeo) [inline src/backends/mirror.ts:83]: prefer a const+ternary if you have 2 values possibles or const+dict lookup if you have more

File: herald.yaml (hunk 2 above)
Comment (destifo) [reply herald.yaml:99]: it doesn't change unless changed manually

File: src/backends/mirror.ts (hunk 3 above)
Comment (destifo) [reply src/backends/mirror.ts:83]: wow, how did I miss that

**Review by zifeo — APPROVED** [review]

**Pull request merged**